src/exceptions.js

```diff
@@ -0,0 +1,119 @@
+import {contains, has} from 'ramda';
+
+export function propTypeErrorHandler(e, props, type) {
+    /*
+     * propType error messages are constructed in
+     * https://github.com/facebook/prop-types/blob/v15.7.2/factoryWithTypeCheckers.js
```

Comment:
interesting, it looks like after this merge, https://github.com/facebook/prop-types/pull/87/files  no one was updating the documentation at all, so unless you dig into the PR storybook, it's a hidden ruby. or is it possible that the strict way is not recommended?

Reply:
Yeah I can't quite figure it out. I get the impression that the team maintaining `prop-types` is pretty overloaded. I think that most of the React team itself has moved over to flow/type script and so doesn't depend on prop-types as much anymore. Not sure though.